python/lsst/drp/tasks/update_visit_summary.py

```diff
@@ -0,0 +1,755 @@
+        storageClass="Background",
+        multiple=True,
+        deferLoad=True,
+    )
+    output_summary_schema = cT.InitOutput(
+        doc="Schema of the output visit summary catalog",
+        name="finalVisitSummary_schema",
+        storageClass="ExposureCatalog",
+    )
+    output_summary_catalog = cT.Output(
+        doc="Visit-level catalog summarizing all image characterizations and calibrations.",
+        name="finalVisitSummary",
+        dimensions=("instrument", "visit"),
+        storageClass="ExposureCatalog",
+    )
+
+    def __init__(self, *, config: UpdateVisitSummaryConfig | None = None):
+        super().__init__(config=config)
+        match self.config.wcs_provider:
+            case "input_summary":
+                self.inputs.remove("wcs_overrides_tract")
+                self.inputs.remove("wcs_overrides_global")
+            case "tract":
+                self.inputs.remove("wcs_overrides_global")
+            case "global":
+                self.inputs.remove("wcs_overrides_tract")
+            case bad:
+                raise ValueError(
+                    f"Invalid value wcs_provider={bad!r}; config was not validated."
+                )
+        match self.config.photo_calib_provider:
+            case "input_summary":
+                self.inputs.remove("photo_calib_overrides_tract")
+                self.inputs.remove("photo_calib_overrides_global")
+            case "tract":
+                self.inputs.remove("photo_calib_overrides_global")
+            case "global":
+                self.inputs.remove("photo_calib_overrides_tract")
+            case bad:
+                raise ValueError(
+                    f"Invalid value photo_calib_provider={bad!r}; config was not validated."
+                )
+        match self.config.background_provider:
+            case "input_summary":
+                self.inputs.remove("background_originals")
+                self.inputs.remove("background_overrides")
+            case "replacement":
+                pass
+            case bad:
+                raise ValueError(
+                    f"Invalid value background_provider={bad!r}; config was not validated."
+                )
+
+
+class UpdateVisitSummaryConfig(
+    PipelineTaskConfig, pipelineConnections=UpdateVisitSummaryConnections
+):
+    """Configuration for UpdateVisitSummaryTask.
+
+    Notes
+    -----
+    The configuration defaults for this task reflect a simple or "least common
+    denominator" pipeline, not the more complete, more sophisticated pipeline
+    we run on the instruments we support best.  The expectation is that the
+    various full pipeline definitions will generally import the simpler
+    definition, so making the defaults correspond to any full pipeline would
+    just lead to the simple pipeline setting them back to the simple-pipeline
+    values and the full pipeline still having to then override them to the
+    full-pipeline values.
+    """
```

Comment:
I was trying to preempt a concern that this task's defaults are not appropriate for the default DRP pipeline, and hence require overrides in that pipeline, since we generally try to avoid that when we can.